ambry-router/src/main/java/com/github/ambry/router/DeleteOperation.java

```diff
@@ -349,6 +359,22 @@ private void onErrorResponse(ReplicaId replicaId, RouterException exception, boo
     }
   }
 
+  /**
+   * Check if we should replicate the blob and retry the operation.
+   * @return true if need to do the retry.
+   */
+  private boolean shouldReplicateBlobAndRetry() {
+    // the conditions to trigger ReplicateBlob and retry the original operation:
+    // 1. the feature is enabled.
+    // 2. and at least one replica returned NOT_FOUND status.
+    // 3. and at least one replica returned successful status.
+    // 4. and error code precedence is over AmbryUnavailable. We don't do replication if there BlobExpired or TooManyRequests.
```

Comment:
if 3 replicas return these values or statuses, does on-demand repl kick in ?
not_found, success & ambry_unavlb

Reply:
We have 9 replicas, when the operation fails, we will check the above conditions to see if ReplicateBlob will help or not.
In your case, don't know what the other 6 replicas returns. But basically yes, if the operation fails, and we have at least 1 success, 1 not_found, we may start on-demand replication.